### ISSUE TITLE
I suspect that the Outgoing portal doesn't wait until the Incoming portal has fully written the file when it tries to pick it up.

### ISSUE THREAD

**Issue by joyd42** [open-automation/switch-portals#23]
I'm having issues with Portals when moving big jobs over flows. These are the messages:

2018-09-21 09:39:34 warning Outgoing Sorteer Preflight2Sorteer unpackJob returned false. Could not unpack. Skipping this file.
2018-09-21 09:39:34 error Outgoing Sorteer Preflight2Sorteer Unarchiving of the job to the temporary folder was unsuccessful.
2018-09-21 09:39:34 assert Outgoing Sorteer Preflight2Sorteer Can't open zip archive: C:/Users/lab9/AppData/Roaming/Enfocus/Switch Server/ScriptData\SwitchPortalsEther\Portals_C-Preflight2Sorteer\_005G8_RO_12015_spandoek_200x100cm_1stuk.pdf.zip

This happens consistently with a 1.1 GB file on Windows. 

I noticed that when the Incoming App is writing the zip file it has a size of 0KB on disk in the SwitchPortalsEther folder. Only when it finished processing the file gets it's full size.

I suspect the Outgoing App already tries to unzip the file when the Incoming App is not done writing it. That would explain the "Can't open zip archive" message.

For now I'll adjust my customers flows by using actual folders to connect flows instead of the Portals App.

Update: Maybe you can use the webhooks and http functionality to let the Outgoing App know when the InComing App finished it's task?

**Comment by Dominick-Peluso-Bose** [comment]
> I suspect the Outgoing App already tries to unzip the file when the Incoming App is not done writing it. That would explain the "Can't open zip archive" message.

That does happen but what is supposed to happen is, the outgoing Portal should skip this file (because it fails to unpack) and then when it checks again, it would succeed the next time.

Does it not resolve itself after the data is written fully?

**Comment by joyd42** [comment]
No, the messages didn't show the job anymore after the portal app. Looks like the jobs vanished. I also couldn't find leftovers in the ether folder.

Sorry for the late reply, I am not getting mail notifications for some reason.
